src/AMSlib/wf/workflow.hpp

```diff
@@ -318,9 +330,16 @@ class AMSWorkflow
       return;
     }
 
-    if (DB && DB->updateModel()) {
-      UQModel->updateModel("");
+    CALIPER(CALI_MARK_BEGIN("UPDATEMODEL");)
+    if (updateModel()) {
+      auto model = DB->getLatestModel();
+      CINFO(Workflow,
+            rId == 0,
+            "Updating surrogate model with %s",
+            model.c_str())
+      UQModel->updateModel(model);
```

Comment:
In a case where multiple models exist in the simulation how would we handle this? Food for thought ...

Reply:
We will have to discuss that use case indeed, and how we select the correct model to use. We could had a model tag in the message that AMSlib receive with the model name and update the right model accordingly.